IPython/core/magics/packaging.py

```diff
@@ -66,7 +66,13 @@ def pip(self, line):
         Usage:
           %pip install [pkgs]
         """
-        python = shlex.quote(sys.executable)
+        python = sys.executable
+        if " " in python:
+            if sys.platform == "win32":
+                python = '"' + python + '"'
+            else:
+                python = shlex.quote(python)
```

Comment:
Branching over `" " in python` isn't necessary here.

```suggestion
        if sys.platform == "win32":
            python = "\"" + python + "\""
        else:
            python = shlex.quote(python)
```